### PR TITLE
Screen orientation: event firing and promise resolving

### PR DESCRIPTION
**Pull request by Johanna-hub**
Relates to https://github.com/w3c/screen-orientation/issues/151

### CODE REVIEW

**Review by marcoscaceres — CHANGES_REQUESTED** [review]

File: screen-orientation/lock-unlock-check.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    const pMustReject = screen.orientation.lock("landscape");
+    const pMustResolve = new Promise(r => {
+      screen.orientation.onchange = async () => {
+        screen.orientation.unlock();
+        r();
+      };
+    });
+    await promise_rejects(t, new TypeError(), pMustReject);
+    await pMustResolve;
+  }, "Event must fire before orientationPendingPromise resolves");
```
Comment (marcoscaceres) [inline screen-orientation/lock-unlock-check.html]: Make sure the tested assertion here is unique too... it should have something to do with .unlock() and orientationPendingPromise.

File: screen-orientation/lock-unlock-check.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script>
+  promise_test(async t => {
+    await test_driver.bless("request full screen", () => {
+      return document.documentElement.requestFullscreen();
+    });
+    const pMustReject = screen.orientation.lock("landscape");
```
[the remark this answers is not in the document]
Comment (Johanna-hub) [reply screen-orientation/lock-unlock-check.html]: I have put this in for now, but once orientation-utils.js is merged, I can just import the `getOppositeOrientation` function here

[the remark this answers is not in the document]
Comment (marcoscaceres) [reply screen-orientation/lock-unlock-check.html]: do you want to send orientation-utils.js separately as a new PR? then we can just merge it in and you can start using it.

[the remark this answers is not in the document]
Comment (Johanna-hub) [reply screen-orientation/lock-unlock-check.html]: Good idea, I'll do it now!

File: screen-orientation/lock-unlock-check.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    await promise_rejects(t, new TypeError(), pMustReject);
+    await pMustResolve;
+  }, "Event must fire before orientationPendingPromise resolves");
+
+  promise_test(async t => {
+    await document.documentElement.requestFullscreen();
+    const pMustReject = screen.orientation.lock("landscape");
+    const pMustResolve = new Promise(r => {
+      screen.orientation.onchange = async () => {
+        screen.orientation.unlock();
```
[the remark this answers is not in the document]
Comment (Johanna-hub) [reply screen-orientation/lock-unlock-check.html]: Yes, that's true!

File: screen-orientation/lock-unlock-check.html (hunk 1 above)
Comment (Johanna-hub) [reply screen-orientation/lock-unlock-check.html]: I've tried to word both, not sure if I've captured it right though.

**Review by marcoscaceres — APPROVED** [review]

File: screen-orientation/lock-unlock-check.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script>
+  import { getOppositeOrientation } from "/screen-orientation/resources/orientation-utils.js";
```
Comment (mathiasbynens) [inline screen-orientation/lock-unlock-check.html]: This is supposed to throw an early SyntaxError, as static `import` only works within `<script type=module>`, so the rest of the `<script>` is essentially ignored. This seems like a bug? cc @leszekswirski

Comment (mathiasbynens) [reply screen-orientation/lock-unlock-check.html]: Filed here https://github.com/web-platform-tests/wpt/issues/17547

Comment (marcoscaceres) [reply screen-orientation/lock-unlock-check.html]: Thanks @mathiasbynens! how weird... I thought we had these working but you are absolutely right that these would be syntax errors :/

**Pull request merged**